### DISCUSSION TITLE
Is it possible to setup the default settings for onboarding in GitHub?

### DISCUSSION THREAD

**Discussion by ChristopherHaws** [renovatebot/renovate#21137] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

For my personal projects, which I host on GitHub, I would like to have renovate update dependencies and automerge the changes without any human intervention. Ideally, I wouldn't even need to run the onboarding process. I would like to configure the default settings for renovate in one place and have those settings apply to all of my personal repositories without adding a `renovate.json` file to every repo.

If the repo does contain a `renovate.json` file, it would get picked up and be used instead of the default config.

### What you tried so far.

- Read through the docs (the onboarding docs are quite confusing)

**Comment by viceice** [comment]
create a shared preset with your default settings. then create a `renovate-config.json` wich references that preset inside the `.github` repo. After this the onboarding willl use that file. so you only need to merge the pr and you can change the config centrally

**Reply by ChristopherHaws** [reply]
I've been using GitHub since 2013 and I never knew that `.github` was a special repo you could use for common, shared items across repos! For anyone wondering, here are the [github docs for the `.github` repo](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/creating-a-default-community-health-file) and here are the renovate docs for [shared config presets](https://docs.renovatebot.com/config-presets/).

I will give that a try and report back! ^_^

Thanks!

**Reply by rarkins** [reply]
I can also help you skip the need for onboarding if you still wish. But in this case it requires a manual backend setting I would do on your behalf

**Reply by viceice** [reply]
this works on all supported renovate platforms. alternatively you can add a `default.json` to `renovate-config` repo which is checked before the mentioned repo.

https://github.com/renovatebot/renovate/blob/6acb760f4ac728f3876bb6857fe5b226f48d1bac/lib/workers/repository/onboarding/branch/config.ts#L26-L79

**Reply by ChristopherHaws** [reply]
@viceice For some reason, renovate keeps trying to add the following config:
```json
{
	"$schema": "https://docs.renovatebot.com/renovate-schema.json",
	"extends": [
		"local>ChristopherHaws/renovate-config"
	]
}
```

But the repo `ChristopherHaws/renovate-config` doesn't exist. Shouldn't it be adding `ChristopherHaws/.github`? I tried renaming and deleting the onboarding PR and Branch, but it created another PR with the same config setting.

**Reply by ChristopherHaws** [reply, marked as answer]
Nevermind, I fixed it. It's because I was using the file name `default.json` instead of `renovate-config.json`

**Reply by ChristopherHaws** [reply]
Is it possible to have renovate run on repos without having to have a `renovate.json` in the repo? If I create a `renovate.json` in my `.github` repo, will renovate use it for all my repos unless the repo has its own `renovate.json`?

**Reply by rarkins** [reply]
Yes, that's the setting I said we'd need to configure in the backend manually though. We would set onboarding=false, requireConfig=optional
